packages/datadog-plugin-playwright/src/index.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const { storage } = require('../../datadog-core')
+const CiPlugin = require('../../dd-trace/src/plugins/ci_plugin')
+
+const {
+  TEST_STATUS,
+  finishAllTraceSpans,
+  getTestParentSpan,
+  getTestSessionCommonTags,
+  getTestSuitePath,
+  TEST_SESSION_ID,
+  TEST_COMMAND,
+  TEST_SUITE_ID,
+  getTestSuiteCommonTags
+} = require('../../dd-trace/src/plugins/util/test')
+const { RESOURCE_NAME } = require('../../../ext/tags')
+const { COMPONENT } = require('../../dd-trace/src/constants')
+
+class PlaywrightPlugin extends CiPlugin {
+  static get name () {
+    return 'playwright'
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    this._testSuites = new Map()
+
+    this.addSub('ci:playwright:session:start', ({ command, frameworkVersion, rootDir }) => {
+      const childOf = getTestParentSpan(this.tracer)
+      this.command = command
+      this.frameworkVersion = frameworkVersion
+      this.rootDir = rootDir
+
+      const testSessionSpanMetadata = getTestSessionCommonTags(command, frameworkVersion)
+      this.testSessionSpan = this.tracer.startSpan('playwright.test_session', {
```

Comment:
this is saved as state in the Plugin... does it means that testSession is unique in whole execution?

Reply:
that's a good question: yes, there's a single test session for the whole execution. There's parallelism in playwright, but it happens at a lower level 😄